diff -dur grub-2.00.orig/Makefile.am grub-2.00/Makefile.am
--- grub-2.00.orig/Makefile.am	2012-06-26 04:07:30.000000000 +0200
+++ grub-2.00/Makefile.am	2012-10-27 16:05:36.203732745 +0200
@@ -122,7 +122,8 @@
 # Install config.h into platformdir
 platform_HEADERS = config.h
 
-pkgdata_DATA += grub-mkconfig_lib
+ulibdir = /lib
+ulib_DATA = grub-mkconfig_lib
 
 
 if COND_i386_coreboot
diff -dur grub-2.00.orig/util/grub-install.in grub-2.00/util/grub-install.in
--- grub-2.00.orig/util/grub-install.in	2012-06-26 23:31:03.000000000 +0200
+++ grub-2.00/util/grub-install.in	2012-10-27 16:04:14.446027790 +0200
@@ -76,7 +76,7 @@
 
 disk_module=unspecified
 
-. "${pkgdatadir}/grub-mkconfig_lib"
+. "/lib/grub-mkconfig_lib"
 
 # Usage: usage
 # Print the usage.
diff -dur grub-2.00.orig/util/grub-kbdcomp.in grub-2.00/util/grub-kbdcomp.in
--- grub-2.00.orig/util/grub-kbdcomp.in	2012-04-05 19:56:36.000000000 +0200
+++ grub-2.00/util/grub-kbdcomp.in	2012-10-27 16:04:14.446027790 +0200
@@ -18,7 +18,7 @@
 export TEXTDOMAIN=@PACKAGE@
 export TEXTDOMAINDIR="@localedir@"
 
-. "${pkgdatadir}/grub-mkconfig_lib"
+. "/lib/grub-mkconfig_lib"
 
 self=`basename $0`
 
diff -dur grub-2.00.orig/util/grub-mkconfig.in grub-2.00/util/grub-mkconfig.in
--- grub-2.00.orig/util/grub-mkconfig.in	2012-10-27 16:01:08.000000000 +0200
+++ grub-2.00/util/grub-mkconfig.in	2012-10-27 16:04:14.449361166 +0200
@@ -46,7 +46,7 @@
 export TEXTDOMAIN=@PACKAGE@
 export TEXTDOMAINDIR="@localedir@"
 
-. "${pkgdatadir}/grub-mkconfig_lib"
+. "/lib/grub-mkconfig_lib"
 
 # Usage: usage
 # Print the usage.
diff -dur grub-2.00.orig/util/grub-mknetdir.in grub-2.00/util/grub-mknetdir.in
--- grub-2.00.orig/util/grub-mknetdir.in	2012-04-11 22:36:24.000000000 +0200
+++ grub-2.00/util/grub-mknetdir.in	2012-10-27 16:04:14.449361166 +0200
@@ -56,7 +56,7 @@
 export TEXTDOMAIN=@PACKAGE@
 export TEXTDOMAINDIR="@localedir@"
 
-. "${pkgdatadir}/grub-mkconfig_lib"
+. "/lib/grub-mkconfig_lib"
 
 # Usage: usage
 # Print the usage.
diff -dur grub-2.00.orig/util/grub-mkrescue.in grub-2.00/util/grub-mkrescue.in
--- grub-2.00.orig/util/grub-mkrescue.in	2012-06-14 19:25:07.000000000 +0200
+++ grub-2.00/util/grub-mkrescue.in	2012-10-27 16:04:14.452694542 +0200
@@ -58,7 +58,7 @@
 
 localedir="@datadir@/locale"
 
-. "${pkgdatadir}/grub-mkconfig_lib"
+. "/lib/grub-mkconfig_lib"
 
 # Usage: usage
 # Print the usage.
diff -dur grub-2.00.orig/util/grub-mkstandalone.in grub-2.00/util/grub-mkstandalone.in
--- grub-2.00.orig/util/grub-mkstandalone.in	2012-04-05 19:54:27.000000000 +0200
+++ grub-2.00/util/grub-mkstandalone.in	2012-10-27 16:04:14.452694542 +0200
@@ -46,7 +46,7 @@
 export TEXTDOMAIN=@PACKAGE@
 export TEXTDOMAINDIR="@localedir@"
 
-. "${pkgdatadir}/grub-mkconfig_lib"
+. "/lib/grub-mkconfig_lib"
 
 # Usage: usage
 # Print the usage.
diff -dur grub-2.00.orig/util/grub-reboot.in grub-2.00/util/grub-reboot.in
--- grub-2.00.orig/util/grub-reboot.in	2012-04-05 19:55:31.000000000 +0200
+++ grub-2.00/util/grub-reboot.in	2012-10-27 16:04:14.456027917 +0200
@@ -40,7 +40,7 @@
 export TEXTDOMAIN=@PACKAGE@
 export TEXTDOMAINDIR="@localedir@"
 
-. "${pkgdatadir}/grub-mkconfig_lib"
+. "/lib/grub-mkconfig_lib"
 
 # Usage: usage
 # Print the usage.
diff -dur grub-2.00.orig/util/grub-set-default.in grub-2.00/util/grub-set-default.in
--- grub-2.00.orig/util/grub-set-default.in	2012-04-05 19:42:05.000000000 +0200
+++ grub-2.00/util/grub-set-default.in	2012-10-27 16:04:14.456027917 +0200
@@ -40,7 +40,7 @@
 export TEXTDOMAIN=@PACKAGE@
 export TEXTDOMAINDIR="@localedir@"
 
-. "${pkgdatadir}/grub-mkconfig_lib"
+. "/lib/grub-mkconfig_lib"
 
 # Usage: usage
 # Print the usage.
diff -dur grub-2.00.orig/util/grub.d/00_header.in grub-2.00/util/grub.d/00_header.in
--- grub-2.00.orig/util/grub.d/00_header.in	2012-05-04 01:04:39.000000000 +0200
+++ grub-2.00/util/grub.d/00_header.in	2012-10-27 16:03:01.041766812 +0200
@@ -27,7 +27,7 @@
 export TEXTDOMAIN=@PACKAGE@
 export TEXTDOMAINDIR="@localedir@"
 
-. "@datadir@/@PACKAGE@/grub-mkconfig_lib"
+. "/lib/grub-mkconfig_lib"
 
 # Do this as early as possible, since other commands might depend on it.
 # (e.g. the `loadfont' command might need lvm or raid modules)
diff -dur grub-2.00.orig/util/grub.d/10_hurd.in grub-2.00/util/grub.d/10_hurd.in
--- grub-2.00.orig/util/grub.d/10_hurd.in	2012-03-04 21:10:04.000000000 +0100
+++ grub-2.00/util/grub.d/10_hurd.in	2012-10-27 16:03:01.045100187 +0200
@@ -24,7 +24,7 @@
 export TEXTDOMAIN=@PACKAGE@
 export TEXTDOMAINDIR="@localedir@"
 
-. "@datadir@/@PACKAGE@/grub-mkconfig_lib"
+. "/lib/grub-mkconfig_lib"
 
 CLASS="--class gnu --class os"
 
diff -dur grub-2.00.orig/util/grub.d/10_illumos.in grub-2.00/util/grub.d/10_illumos.in
--- grub-2.00.orig/util/grub.d/10_illumos.in	2012-03-04 20:03:38.000000000 +0100
+++ grub-2.00/util/grub.d/10_illumos.in	2012-10-27 16:03:01.045100187 +0200
@@ -20,7 +20,7 @@
 prefix="@prefix@"
 exec_prefix="@exec_prefix@"
 datarootdir="@datarootdir@"
-. "@datadir@/@PACKAGE@/grub-mkconfig_lib"
+. "/lib/grub-mkconfig_lib"
 
 export TEXTDOMAIN=@PACKAGE@
 export TEXTDOMAINDIR="@localedir@"
diff -dur grub-2.00.orig/util/grub.d/10_kfreebsd.in grub-2.00/util/grub.d/10_kfreebsd.in
--- grub-2.00.orig/util/grub.d/10_kfreebsd.in	2012-03-04 22:02:30.000000000 +0100
+++ grub-2.00/util/grub.d/10_kfreebsd.in	2012-10-27 16:03:01.045100187 +0200
@@ -20,7 +20,7 @@
 prefix="@prefix@"
 exec_prefix="@exec_prefix@"
 datarootdir="@datarootdir@"
-. "@datadir@/@PACKAGE@/grub-mkconfig_lib"
+. "/lib/grub-mkconfig_lib"
 
 export TEXTDOMAIN=@PACKAGE@
 export TEXTDOMAINDIR="@localedir@"
diff -dur grub-2.00.orig/util/grub.d/10_linux.in grub-2.00/util/grub.d/10_linux.in
--- grub-2.00.orig/util/grub.d/10_linux.in	2012-10-27 16:01:08.000000000 +0200
+++ grub-2.00/util/grub.d/10_linux.in	2012-10-27 16:03:01.048433562 +0200
@@ -21,7 +21,7 @@
 exec_prefix="@exec_prefix@"
 datarootdir="@datarootdir@"
 
-. "@datadir@/@PACKAGE@/grub-mkconfig_lib"
+. "/lib/grub-mkconfig_lib"
 
 export TEXTDOMAIN=@PACKAGE@
 export TEXTDOMAINDIR="@localedir@"
diff -dur grub-2.00.orig/util/grub.d/10_netbsd.in grub-2.00/util/grub.d/10_netbsd.in
--- grub-2.00.orig/util/grub.d/10_netbsd.in	2012-03-04 20:47:35.000000000 +0100
+++ grub-2.00/util/grub.d/10_netbsd.in	2012-10-27 16:03:01.048433562 +0200
@@ -20,7 +20,7 @@
 prefix="@prefix@"
 exec_prefix="@exec_prefix@"
 datarootdir="@datarootdir@"
-. "@datadir@/@PACKAGE@/grub-mkconfig_lib"
+. "/lib/grub-mkconfig_lib"
 
 export TEXTDOMAIN=@PACKAGE@
 export TEXTDOMAINDIR="@localedir@"
diff -dur grub-2.00.orig/util/grub.d/10_windows.in grub-2.00/util/grub.d/10_windows.in
--- grub-2.00.orig/util/grub.d/10_windows.in	2012-03-04 22:11:43.000000000 +0100
+++ grub-2.00/util/grub.d/10_windows.in	2012-10-27 16:03:01.048433562 +0200
@@ -24,7 +24,7 @@
 export TEXTDOMAIN=@PACKAGE@
 export TEXTDOMAINDIR="@localedir@"
 
-. "@datadir@/@PACKAGE@/grub-mkconfig_lib"
+. "/lib/grub-mkconfig_lib"
 
 case "`uname 2>/dev/null`" in
   CYGWIN*)  ;;
diff -dur grub-2.00.orig/util/grub.d/10_xnu.in grub-2.00/util/grub.d/10_xnu.in
--- grub-2.00.orig/util/grub.d/10_xnu.in	2012-05-29 12:31:03.000000000 +0200
+++ grub-2.00/util/grub.d/10_xnu.in	2012-10-27 16:03:01.048433562 +0200
@@ -24,7 +24,7 @@
 export TEXTDOMAIN=@PACKAGE@
 export TEXTDOMAINDIR="@localedir@"
 
-. "@datadir@/@PACKAGE@/grub-mkconfig_lib"
+. "/lib/grub-mkconfig_lib"
 
 osx_entry() {
     if [ x$2 = x32 ]; then
diff -dur grub-2.00.orig/util/grub.d/20_linux_xen.in grub-2.00/util/grub.d/20_linux_xen.in
--- grub-2.00.orig/util/grub.d/20_linux_xen.in	2012-06-03 21:57:42.000000000 +0200
+++ grub-2.00/util/grub.d/20_linux_xen.in	2012-10-27 16:03:01.051766938 +0200
@@ -21,7 +21,7 @@
 exec_prefix="@exec_prefix@"
 datarootdir="@datarootdir@"
 
-. "@datadir@/@PACKAGE@/grub-mkconfig_lib"
+. "/lib/grub-mkconfig_lib"
 
 export TEXTDOMAIN=@PACKAGE@
 export TEXTDOMAINDIR="@localedir@"
diff -dur grub-2.00.orig/util/grub.d/30_os-prober.in grub-2.00/util/grub.d/30_os-prober.in
--- grub-2.00.orig/util/grub.d/30_os-prober.in	2012-03-04 21:52:03.000000000 +0100
+++ grub-2.00/util/grub.d/30_os-prober.in	2012-10-27 16:03:01.051766938 +0200
@@ -24,7 +24,7 @@
 export TEXTDOMAIN=@PACKAGE@
 export TEXTDOMAINDIR="@localedir@"
 
-. "@datadir@/@PACKAGE@/grub-mkconfig_lib"
+. "/lib/grub-mkconfig_lib"
 
 if [ "x${GRUB_DISABLE_OS_PROBER}" = "xtrue" ]; then
   exit 0
