--- grub-1.96/10_linux~	2009-01-31 12:01:43.139168036 +0100
+++ grub-1.96/10_linux	2009-01-31 12:24:29.307205454 +0100
@@ -92,8 +92,8 @@
   alt_version=`echo $version | sed -e "s,\.old$,,g"`
 
   initrd=
-  for i in "initrd.img-${version}" "initrd-${version}.img" \
-	   "initrd.img-${alt_version}" "initrd-${alt_version}.img"; do
+  for i in "initrd-${version}" "initrd-${version}.gz" \
+	   "initrd-${alt_version}" "initrd-${alt_version}.gz"; do
     if test -e "${dirname}/${i}" ; then
       initrd="$i"
       break
