diff -dur -x '*~' grub-2.00.orig/grub-core/Makefile.core.def grub-2.00/grub-core/Makefile.core.def
--- grub-2.00.orig/grub-core/Makefile.core.def	2013-12-10 13:20:26.000000000 +0100
+++ grub-2.00/grub-core/Makefile.core.def	2013-12-10 13:23:01.000000000 +0100
@@ -1848,3 +1848,4 @@
   enable = i386;
 };
 
+#include Makefile.gcry.def
diff -dur -x '*~' grub-2.00.orig/grub-core/Makefile.in grub-2.00/grub-core/Makefile.in
--- grub-2.00.orig/grub-core/Makefile.in	2012-06-26 14:12:27.000000000 +0200
+++ grub-2.00/grub-core/Makefile.in	2013-12-10 13:22:48.000000000 +0100
@@ -70505,7 +70505,7 @@
 .PRECIOUS: $(top_srcdir)/grub-core/Makefile.core.am
 $(top_srcdir)/grub-core/Makefile.core.am: $(top_srcdir)/grub-core/Makefile.core.def $(top_srcdir)/grub-core/Makefile.gcry.def $(top_srcdir)/Makefile.tpl
 	if [ "x$$GRUB_CONTRIB" != x ]; then echo "You need to run ./autogen.sh manually." >&2; exit 1; fi
-	autogen -T $(top_srcdir)/Makefile.tpl $(top_srcdir)/grub-core/Makefile.core.def $(top_srcdir)/grub-core/Makefile.gcry.def | sed -e '/^$$/{N;/^\\n$$/D;}' > $@.new || (rm -f $@.new; exit 1)
+	autogen -T $(top_srcdir)/Makefile.tpl $(top_srcdir)/grub-core/Makefile.core.def | sed -e '/^$$/{N;/^\\n$$/D;}' > $@.new || (rm -f $@.new; exit 1)
 	mv $@.new $@
 
 # gentrigtables
--- grub-2.00.orig/conf/Makefile.common	2013-12-10 13:20:26.000000000 +0100
+++ grub-2.00/conf/Makefile.common	2013-12-10 13:44:26.000000000 +0100
@@ -174,5 +174,5 @@
 .PRECIOUS: $(top_srcdir)/grub-core/Makefile.core.am
 $(top_srcdir)/grub-core/Makefile.core.am: $(top_srcdir)/grub-core/Makefile.core.def $(top_srcdir)/grub-core/Makefile.gcry.def $(top_srcdir)/Makefile.tpl
 	if [ "x$$GRUB_CONTRIB" != x ]; then echo "You need to run ./autogen.sh manually." >&2; exit 1; fi
-	autogen -T $(top_srcdir)/Makefile.tpl $(top_srcdir)/grub-core/Makefile.core.def $(top_srcdir)/grub-core/Makefile.gcry.def | sed -e '/^$$/{N;/^\\n$$/D;}' > $@.new || (rm -f $@.new; exit 1)
+	autogen -T $(top_srcdir)/Makefile.tpl $(top_srcdir)/grub-core/Makefile.core.def | sed -e '/^$$/{N;/^\\n$$/D;}' > $@.new || (rm -f $@.new; exit 1)
 	mv $@.new $@
--- grub-2.00.orig/configure.ac	2013-12-10 13:32:13.000000000 +0100
+++ grub-2.00/configure.ac	2013-12-10 13:32:19.000000000 +0100
@@ -1169,6 +1169,11 @@
 AC_CONFIG_FILES([util/bash-completion.d/Makefile])
 AC_CONFIG_FILES([stamp-h], [echo timestamp > stamp-h])
 AC_CONFIG_FILES([config.h])
+AC_CONFIG_FILES([grub-core/gdb_grub])
+AC_CONFIG_FILES([grub-core/genmod.sh])
+AC_CONFIG_FILES([grub-core/gensyminfo.sh])
+AC_CONFIG_FILES([grub-core/gmodule.pl])
+AC_CONFIG_FILES([grub-core/modinfo.sh])
 
 AC_OUTPUT
 [

