--- grub-1.99/Makefile.am~	2011-04-16 15:29:09.000000000 +0200
+++ grub-1.99/Makefile.am	2011-09-08 15:18:49.286379677 +0200
@@ -101,7 +101,8 @@
 # Install config.h into platformdir
 platform_HEADERS = config.h
 
-pkglib_DATA += grub-mkconfig_lib
+ulibdir = /lib
+ulib_DATA = grub-mkconfig_lib
 pkglib_DATA += update-grub_lib
 
 
--- grub-1.97.1.shelllib/util/grub.d/00_header.in	2009-11-09 17:48:16.000000000 +0200
+++ grub-1.97.1/util/grub.d/00_header.in	2009-11-17 12:34:23.164322142 +0200
@@ -23,7 +23,7 @@
 libdir=@libdir@
 grub_prefix=`echo /boot/grub | sed ${transform}`
 
-. ${libdir}/grub/grub-mkconfig_lib
+. /lib/grub-mkconfig_lib
 
 # Do this as early as possible, since other commands might depend on it.
 # (e.g. the `loadfont' command might need lvm or raid modules)
--- grub-1.97.1.shelllib/util/grub.d/10_hurd.in	2009-11-09 17:48:16.000000000 +0200
+++ grub-1.97.1/util/grub.d/10_hurd.in	2009-11-17 12:34:23.160988747 +0200
@@ -19,7 +19,7 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
-. ${libdir}/grub/grub-mkconfig_lib
+. /lib/grub-mkconfig_lib
 
 if [ "x${GRUB_DISTRIBUTOR}" = "x" ] ; then
   OS=GNU
--- grub-1.97.1.shelllib/util/grub.d/10_linux.in	2009-11-17 12:34:05.445365474 +0200
+++ grub-1.97.1/util/grub.d/10_linux.in	2009-11-17 12:34:23.160988747 +0200
@@ -19,7 +19,7 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
-. ${libdir}/grub/grub-mkconfig_lib
+. /lib/grub-mkconfig_lib
 
 if [ "x${GRUB_DISTRIBUTOR}" = "x" ] ; then
   OS=GNU/Linux
--- grub-1.97.1.shelllib/util/grub.d/10_windows.in	2009-11-09 17:48:16.000000000 +0200
+++ grub-1.97.1/util/grub.d/10_windows.in	2009-11-17 12:34:23.160988747 +0200
@@ -19,7 +19,7 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
-. ${libdir}/grub/grub-mkconfig_lib
+. /lib/grub-mkconfig_lib
 
 case "`uname 2>/dev/null`" in
   CYGWIN*)  ;;
--- grub-1.97.1.shelllib/util/grub.d/30_os-prober.in	2009-11-09 17:48:16.000000000 +0200
+++ grub-1.97.1/util/grub.d/30_os-prober.in	2009-11-17 12:34:23.164322142 +0200
@@ -20,7 +20,7 @@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
 
-. ${libdir}/grub/grub-mkconfig_lib
+. /lib/grub-mkconfig_lib
 
 if [ "x${GRUB_DISABLE_OS_PROBER}" = "xtrue" ]; then
   exit 0
--- grub-1.97.1.shelllib/util/grub-mkconfig.in	2009-11-17 12:34:05.448699048 +0200
+++ grub-1.97.1/util/grub-mkconfig.in	2009-11-17 12:34:23.167655247 +0200
@@ -73,7 +73,7 @@
     esac
 done
 
-. ${libdir}/grub/grub-mkconfig_lib
+. /lib/grub-mkconfig_lib
 
 if [ "x$EUID" = "x" ] ; then
   EUID=`id -u`
--- grub-1.97.1.shelllib/util/update-grub_lib.in	2009-11-09 17:48:16.000000000 +0200
+++ grub-1.97.1/util/update-grub_lib.in	2009-11-17 12:34:23.157655729 +0200
@@ -18,6 +18,6 @@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
 
-. ${libdir}/grub/grub-mkconfig_lib
+. /lib/grub-mkconfig_lib
 
 grub_warn "update-grub_lib is deprecated, use grub-mkconfig_lib instead"
--- grub-1.99/util/grub.d/20_linux_xen.in~	2011-04-13 21:20:29.000000000 +0200
+++ grub-1.99/util/grub.d/20_linux_xen.in	2011-09-08 17:52:27.113229582 +0200
@@ -21,7 +21,7 @@
 exec_prefix=@exec_prefix@
 bindir=@bindir@
 libdir=@libdir@
-. ${libdir}/grub/grub-mkconfig_lib
+. /lib/grub-mkconfig_lib
 
 export TEXTDOMAIN=@PACKAGE@
 export TEXTDOMAINDIR=@localedir@

