diff -dur grub-2.00.orig/conf/Makefile.common grub-2.00/conf/Makefile.common
--- grub-2.00.orig/conf/Makefile.common	2012-05-27 18:16:05.000000000 +0200
+++ grub-2.00/conf/Makefile.common	2012-10-27 15:00:48.537431538 +0200
@@ -99,7 +99,7 @@
 # Other variables
 
 grubconfdir = $(sysconfdir)/grub.d
-platformdir = $(pkglibdir)/$(target_cpu)-$(platform)
+platformdir = $(pkglibdir)
 starfielddir = $(pkgdatadir)/themes/starfield
 
 CFLAGS_GCRY = -Wno-error -Wno-missing-field-initializers
diff -dur grub-2.00.orig/util/grub-install.in grub-2.00/util/grub-install.in
--- grub-2.00.orig/util/grub-install.in	2012-06-26 23:31:03.000000000 +0200
+++ grub-2.00/util/grub-install.in	2012-10-27 15:03:21.133028965 +0200
@@ -522,23 +522,6 @@
     device_map=
 fi
 
-# Copy the GRUB images to the GRUB directory.
-for file in "${grubdir}"/*.mod "${grubdir}"/*.lst "${grubdir}"/*.img "${grubdir}"/efiemu??.o "${grubdir}"/${grub_modinfo_target_cpu}-$grub_modinfo_platform/*.mod "${grubdir}"/${grub_modinfo_target_cpu}-$grub_modinfo_platform/*.lst "${grubdir}"/${grub_modinfo_target_cpu}-$grub_modinfo_platform/*.img "${grubdir}"/${grub_modinfo_target_cpu}-$grub_modinfo_platform/efiemu??.o; do
-    if test -f "$file" && [ "`basename $file`" != menu.lst ]; then
-	rm -f "$file" || exit 1
-    fi
-done
-for file in "${source_dir}"/*.mod "${source_dir}"/*.lst; do
-    cp -f "$file" "${grubdir}/${grub_modinfo_target_cpu}-$grub_modinfo_platform" || exit 1
-done
-if [ "${grub_modinfo_target_cpu}-${grub_modinfo_platform}" = "i386-pc" ] || [ "${grub_modinfo_target_cpu}-${grub_modinfo_platform}" = "sparc64-ieee1275" ] ; then
-    for file in "${source_dir}"/*.img "${source_dir}"/efiemu??.o; do
-	if test -f "$file"; then
-	    cp -f "$file" "${grubdir}/${grub_modinfo_target_cpu}-$grub_modinfo_platform" || exit 1
-	fi
-    done
-fi
-
 # Copy gettext files
 mkdir -p "${grubdir}"/locale/
 for dir in "${localedir}"/*; do
