--- grub-1.99~rc2/conf/Makefile.common~	2011-09-08 12:39:42.000000000 +0200
+++ grub-1.99~rc2/conf/Makefile.common	2011-09-08 13:12:11.194868113 +0200
@@ -100,7 +100,7 @@
 
 grubconfdir = /lib/grub.d
 grubdatadir = $(datadir)/`echo @PACKAGE_TARNAME@ | sed '$(transform)'`
-platformdir = $(pkglibrootdir)/$(target_cpu)-$(platform)
+platformdir = $(pkglibrootdir)
 
 CFLAGS_GCRY = -Wno-error -Wno-missing-field-initializers
 CPPFLAGS_GCRY = -I$(top_srcdir)/grub-core/lib/libgcrypt_wrap
--- grub-1.99~rc2/util/grub-install.in~	2011-04-03 15:36:21.000000000 +0200
+++ grub-1.99~rc2/util/grub-install.in	2011-09-06 22:18:41.574849650 +0200
@@ -31,7 +31,7 @@
 target_cpu=@target_cpu@
 platform=@platform@
 host_os=@host_os@
-pkglibdir="${libdir}/`echo ${PACKAGE_TARNAME}/${target_cpu}-${platform} | sed ${transform}`"
+pkglibdir="${libdir}/`echo ${PACKAGE_TARNAME} | sed ${transform}`"
 localedir="@datadir@/locale"
 
 self="`basename $0`"
@@ -431,23 +431,6 @@
     exit 1
 fi
 
-# Copy the GRUB images to the GRUB directory.
-for file in "${grubdir}"/*.mod "${grubdir}"/*.lst "${grubdir}"/*.img "${grubdir}"/efiemu??.o; do
-    if test -f "$file" && [ "`basename $file`" != menu.lst ]; then
-	rm -f "$file" || exit 1
-    fi
-done
-for file in "${pkglibdir}"/*.mod "${pkglibdir}"/*.lst; do
-    cp -f "$file" "${grubdir}" || exit 1
-done
-if [ "${target_cpu}-${platform}" = "i386-pc" ] || [ "${target_cpu}-${platform}" = "sparc64-ieee1275" ] ; then
-    for file in "${pkglibdir}"/*.img "${pkglibdir}"/efiemu??.o; do
-	if test -f "$file"; then
-	    cp -f "$file" "${grubdir}" || exit 1
-	fi
-    done
-fi
-
 # Copy gettext files
 mkdir -p "${grubdir}"/locale/
 for dir in "${localedir}"/*; do

