diff -dur grub-2.00.orig/util/grub-mkconfig.in grub-2.00/util/grub-mkconfig.in
--- grub-2.00.orig/util/grub-mkconfig.in	2012-10-27 15:05:15.000000000 +0200
+++ grub-2.00/util/grub-mkconfig.in	2012-10-27 15:14:32.787243346 +0200
@@ -221,11 +236,16 @@
 if test "x${grub_cfg}" != "x"; then
   rm -f "${grub_cfg}.new"
   oldumask=$(umask); umask 077
+  # open fd &3 for diagnostic messages
+  exec 3>&1
   exec > "${grub_cfg}.new"
   umask $oldumask
+else
+  # open fd &3 for diagnostic messages
+  exec 3>&2
 fi
-gettext "Generating grub.cfg ..." >&2
-echo >&2
+gettext "Generating grub.cfg ..." >&3
+echo >&3
 
 cat << EOF
 #
@@ -259,12 +279,15 @@
     gettext_printf "Syntax errors are detected in generated GRUB config file.
 Ensure that there are no errors in /etc/sysconfig/grub
 and /etc/grub.d/* files or please file a bug report with
-%s file attached." "${grub_cfg}.new" >&2
+%s file attached." "${grub_cfg}.new" >&3
   else
     # none of the children aborted with error, install the new grub.cfg
     mv -f ${grub_cfg}.new ${grub_cfg}
   fi
 fi
 
-gettext "done" >&2
-echo >&2
+gettext "done" >&3
+echo >&3
+
+# close diagnostic stream
+exec 3>&-
diff -dur grub-2.00.orig/util/grub.d/00_header.in grub-2.00/util/grub.d/00_header.in
--- grub-2.00.orig/util/grub.d/00_header.in	2012-05-04 01:04:39.000000000 +0200
+++ grub-2.00/util/grub.d/00_header.in	2012-10-27 15:16:00.665356307 +0200
@@ -219,7 +219,7 @@
 if [ "x$gfxterm" = x1 ]; then
     if [ "x$GRUB_THEME" != x ] && [ -f "$GRUB_THEME" ] \
 	&& is_path_readable_by_grub "$GRUB_THEME"; then
-	gettext_printf "Found theme: %s\n" "$GRUB_THEME" >&2
+	gettext_printf "Found theme: %s\n" "$GRUB_THEME" >&3
 
 	prepare_grub_to_access_device `${grub_probe} --target=device "$GRUB_THEME"`
 	cat << EOF
@@ -255,12 +255,12 @@
 EOF
     elif [ "x$GRUB_BACKGROUND" != x ] && [ -f "$GRUB_BACKGROUND" ] \
 	    && is_path_readable_by_grub "$GRUB_BACKGROUND"; then
-	gettext_printf "Found background: %s\n" "$GRUB_BACKGROUND" >&2
+	gettext_printf "Found background: %s\n" "$GRUB_BACKGROUND" >&3
 	case "$GRUB_BACKGROUND" in 
 	    *.png)         reader=png ;;
 	    *.tga)         reader=tga ;;
 	    *.jpg|*.jpeg)  reader=jpeg ;;
-	    *)             gettext "Unsupported image format" >&2; echo >&2; exit 1 ;;
+	    *)             gettext "Unsupported image format" >&3; echo >&3; exit 1 ;;
 	esac
 	prepare_grub_to_access_device `${grub_probe} --target=device "$GRUB_BACKGROUND"`
 	cat << EOF
diff -dur grub-2.00.orig/util/grub.d/10_hurd.in grub-2.00/util/grub.d/10_hurd.in
--- grub-2.00.orig/util/grub.d/10_hurd.in	2012-03-04 21:10:04.000000000 +0100
+++ grub-2.00/util/grub.d/10_hurd.in	2012-10-27 15:16:00.665356307 +0200
@@ -45,8 +45,8 @@
     basename=`basename $i`
     dirname=`dirname $i`
     rel_dirname=`make_system_path_relative_to_its_root $dirname`
-    gettext_printf "Found GNU Mach: %s" "$i" >&2
-    echo >&2
+    gettext_printf "Found GNU Mach: %s" "$i" >&3
+    echo >&3
     kernels="${kernels} ${rel_dirname}/${basename}"
     at_least_one=true
   fi
@@ -60,8 +60,8 @@
 
 for i in /hurd/${hurd_fs}.static /hurd/exec ; do
   if test -e "$i" ; then
-    gettext_printf "Found Hurd module: %s" "$i" >&2
-    echo >&2
+    gettext_printf "Found Hurd module: %s" "$i" >&3
+    echo >&3
     at_least_one=true
   else
     all_of_them=false
@@ -74,8 +74,8 @@
 fi
 
 if ${all_of_them} && test -e /lib/ld.so.1 ; then : ; else
-  gettext "Some Hurd stuff found, but not enough to boot." >&2
-  echo >&2
+  gettext "Some Hurd stuff found, but not enough to boot." >&3
+  echo >&3
   exit 1
 fi
 
diff -dur grub-2.00.orig/util/grub.d/10_kfreebsd.in grub-2.00/util/grub.d/10_kfreebsd.in
--- grub-2.00.orig/util/grub.d/10_kfreebsd.in	2012-03-04 22:02:30.000000000 +0100
+++ grub-2.00/util/grub.d/10_kfreebsd.in	2012-10-27 15:16:00.668689695 +0200
@@ -158,7 +158,7 @@
 
 while [ "x$list" != "x" ] ; do
   kfreebsd=`version_find_latest $list`
-  gettext_printf "Found kernel of FreeBSD: %s\n" "$kfreebsd" >&2
+  gettext_printf "Found kernel of FreeBSD: %s\n" "$kfreebsd" >&3
   basename=`basename $kfreebsd`
   dirname=`dirname $kfreebsd`
   rel_dirname=`make_system_path_relative_to_its_root $dirname`
@@ -207,7 +207,7 @@
     fi
   done
   if test -n "${module_dir}" ; then
-    gettext_printf "Found kernel module directory: %s\n" "${module_dir}" >&2
+    gettext_printf "Found kernel module directory: %s\n" "${module_dir}" >&3
     module_dir_rel=$(make_system_path_relative_to_its_root $module_dir)
   fi
 
diff -dur grub-2.00.orig/util/grub.d/10_linux.in grub-2.00/util/grub.d/10_linux.in
--- grub-2.00.orig/util/grub.d/10_linux.in	2012-10-27 15:05:15.000000000 +0200
+++ grub-2.00/util/grub.d/10_linux.in	2012-10-27 15:16:00.668689695 +0200
@@ -182,7 +182,7 @@
 is_first_entry=true
 while [ "x$list" != "x" ] ; do
   linux=`version_find_latest $list`
-  gettext_printf "Found linux image: %s\n" "$linux" >&2
+  gettext_printf "Found linux image: %s\n" "$linux" >&3
   basename=`basename $linux`
   dirname=`dirname $linux`
   rel_dirname=`make_system_path_relative_to_its_root $dirname`
@@ -220,7 +220,7 @@
   fi
 
   if test -n "${initrd}" ; then
-    gettext_printf "Found initrd image: %s\n" "${dirname}/${initrd}" >&2
+    gettext_printf "Found initrd image: %s\n" "${dirname}/${initrd}" >&3
   elif test -z "${initramfs}" ; then
     # "UUID=" and "ZFS=" magic is parsed by initrd or initramfs.  Since there's
     # no initrd or builtin initramfs, it can't work here.
diff -dur grub-2.00.orig/util/grub.d/10_netbsd.in grub-2.00/util/grub.d/10_netbsd.in
--- grub-2.00.orig/util/grub.d/10_netbsd.in	2012-03-04 20:47:35.000000000 +0100
+++ grub-2.00/util/grub.d/10_netbsd.in	2012-10-27 15:16:00.668689695 +0200
@@ -155,7 +155,7 @@
     continue
   fi
 
-  gettext_printf "Found NetBSD kernel: %s\n" "$k" >&2
+  gettext_printf "Found NetBSD kernel: %s\n" "$k" >&3
 
   if [ "x$is_first_entry" = xtrue ]; then
       netbsd_entry "knetbsd"   "$k" simple "${GRUB_CMDLINE_NETBSD_DEFAULT}"
diff -dur grub-2.00.orig/util/grub.d/10_windows.in grub-2.00/util/grub.d/10_windows.in
--- grub-2.00.orig/util/grub.d/10_windows.in	2012-03-04 22:11:43.000000000 +0100
+++ grub-2.00/util/grub.d/10_windows.in	2012-10-27 15:16:00.668689695 +0200
@@ -82,7 +82,7 @@
   # Get boot device.
   dev=`${grub_probe} -t device "$dir" 2>/dev/null` || continue
 
-  gettext_printf "Found %s on %s (%s)\n" "$OS" "$drv" "$dev" >&2
+  gettext_printf "Found %s on %s (%s)\n" "$OS" "$drv" "$dev" >&3
   cat << EOF
 menuentry '$(echo "$OS" | grub_quote)' \$menuentry_id_option '$osid-$(grub_get_device_id "${dev}")' {
 EOF
diff -dur grub-2.00.orig/util/grub.d/20_linux_xen.in grub-2.00/util/grub.d/20_linux_xen.in
--- grub-2.00.orig/util/grub.d/20_linux_xen.in	2012-06-03 21:57:42.000000000 +0200
+++ grub-2.00/util/grub.d/20_linux_xen.in	2012-10-27 15:16:00.672023083 +0200
@@ -203,7 +203,7 @@
     fi
     while [ "x$list" != "x" ] ; do
 	linux=`version_find_latest $list`
-	gettext_printf "Found linux image: %s\n" "$linux" >&2
+	gettext_printf "Found linux image: %s\n" "$linux" >&3
 	basename=`basename $linux`
 	dirname=`dirname $linux`
 	rel_dirname=`make_system_path_relative_to_its_root $dirname`
@@ -226,7 +226,7 @@
 	    fi
 	done
 	if test -n "${initrd}" ; then
-	    gettext_printf "Found initrd image: %s\n" "${dirname}/${initrd}" >&2
+	    gettext_printf "Found initrd image: %s\n" "${dirname}/${initrd}" >&3
 	else
     # "UUID=" magic is parsed by initrds.  Since there's no initrd, it can't work here.
 	    linux_root_device_thisversion=${GRUB_DEVICE}
diff -dur grub-2.00.orig/util/grub.d/30_os-prober.in grub-2.00/util/grub.d/30_os-prober.in
--- grub-2.00.orig/util/grub.d/30_os-prober.in	2012-03-04 21:52:03.000000000 +0100
+++ grub-2.00/util/grub.d/30_os-prober.in	2012-10-27 15:16:00.675356471 +0200
@@ -117,7 +117,7 @@
     LONGNAME="${LABEL}"
   fi
 
-  gettext_printf "Found %s on %s\n" "${LONGNAME}" "${DEVICE}" >&2
+  gettext_printf "Found %s on %s\n" "${LONGNAME}" "${DEVICE}" >&3
 
   case ${BOOT} in
     chain)
@@ -267,7 +267,7 @@
     *)
       echo -n "  "
       # TRANSLATORS: %s is replaced by OS name.
-      gettext_printf "%s is not yet supported by grub-mkconfig.\n" "${LONGNAME}" >&2
+      gettext_printf "%s is not yet supported by grub-mkconfig.\n" "${LONGNAME}" >&3
     ;;
   esac
 done
