diff -dur -x '*.orig' grub-2.00.git20131218.orig/util/grub-mkconfig.in grub-2.00.git20131218/util/grub-mkconfig.in
--- grub-2.00.git20131218.orig/util/grub-mkconfig.in	2013-12-18 14:20:33.000000000 +0100
+++ grub-2.00.git20131218/util/grub-mkconfig.in	2013-12-18 14:21:31.000000000 +0100
@@ -138,6 +138,27 @@
 
 # Device containing our userland.  Typically used for root= parameter.
 GRUB_DEVICE="`${grub_probe} --target=device "$GRUB_ROOT"/`"
+
+# Rewrite to sane LVM node (Code from PLD Linux geninitrd):
+# /dev/mapper/sys-rootfs -> /dev/sys/rootfs
+# /dev/mapper/blodnatt-blah--bleh -> /dev/blodnatt/blah-bleh
+# /dev/mapper/vg--meaw-root -> /dev/vg-meaw/root
+case "$GRUB_DEVICE" in
+	/dev/mapper/*-*)
+		DM_REAL_DEVICE="$(readlink -f "$GRUB_DEVICE")"
+		DM_UUID="$(cat /sys/block/${DM_REAL_DEVICE#/dev/}/dm/uuid)"
+		case "$DM_UUID" in
+			LVM-*)
+				# change "--" to / (as "/" is impossible in LV name)
+				local dev=$(awk -vdev="${GRUB_DEVICE#/dev/mapper/}" 'BEGIN{gsub(/--/, "/", dev); print dev}')
+				local VG=$(awk -vdev="$dev" 'BEGIN{split(dev, v, "-"); gsub("/", "-", v[1]); print v[1]}')
+				local LV=$(awk -vdev="$dev" 'BEGIN{split(dev, v, "-"); gsub("/", "-", v[2]); print v[2]}')
+				GRUB_DEVICE=/dev/$VG/$LV
+				;;
+		esac
+	;;
+esac
+
 GRUB_DEVICE_UUID="`${grub_probe} --device ${GRUB_DEVICE} --target=fs_uuid 2> /dev/null`" || true
 
 # Device containing our /boot partition.  Usually the same as GRUB_DEVICE.
