diff -dur grub-2.00.orig/util/grub-mkconfig_lib.in grub-2.00/util/grub-mkconfig_lib.in
--- grub-2.00.orig/util/grub-mkconfig_lib.in	2012-06-27 17:27:53.000000000 +0200
+++ grub-2.00/util/grub-mkconfig_lib.in	2012-10-27 14:55:58.226216374 +0200
@@ -178,6 +178,7 @@
     case "$1" in
       *.dpkg-*) return 1 ;; # debian dpkg
       *.rpmsave|*.rpmnew) return 1 ;;
+      *.bak|*~|.*.swp) return 1 ;;    # backup and swap files from editors
       README*|*/README*)  return 1 ;; # documentation
     esac
   else
