diff -dur -x '*~' grub-2.00.orig/configure.ac grub-2.00/configure.ac
--- grub-2.00.orig/configure.ac	2012-10-28 09:47:24.000000000 +0100
+++ grub-2.00/configure.ac	2012-10-28 09:48:06.438240771 +0100
@@ -940,7 +940,7 @@
 
 if test x"$starfield_excuse" = x; then
    for ext in pcf pcf.gz bdf bdf.gz ttf ttf.gz; do
-     for dir in . /usr/src /usr/share/fonts/X11/misc /usr/share/fonts/truetype/ttf-dejavu /usr/share/fonts/dejavu /usr/share/fonts/truetype; do
+     for dir in . /usr/share/fonts/TTF /usr/src /usr/share/fonts/X11/misc /usr/share/fonts/truetype/ttf-dejavu /usr/share/fonts/dejavu /usr/share/fonts/truetype; do
         if test -f "$dir/DejaVuSans.$ext"; then
           DJVU_FONT_SOURCE="$dir/DejaVuSans.$ext"
           break 2
