diff -dur -x '*.orig' grub-2.00.git20131218.orig/util/grub.d/10_linux.in grub-2.00.git20131218/util/grub.d/10_linux.in
--- grub-2.00.git20131218.orig/util/grub.d/10_linux.in	2013-12-18 14:41:17.000000000 +0100
+++ grub-2.00.git20131218/util/grub.d/10_linux.in	2013-12-18 14:42:46.000000000 +0100
@@ -54,9 +54,9 @@
 CLASS="--class gnu-linux --class gnu --class os"
 
 if [ "x${GRUB_DISTRIBUTOR}" = "x" ] ; then
-  OS=GNU/Linux
+  OS="$(. /etc/os-release ; echo "$NAME $VERSION")"
 else
-  OS="${GRUB_DISTRIBUTOR} GNU/Linux"
+  OS="${GRUB_DISTRIBUTOR}"
   CLASS="--class $(echo ${GRUB_DISTRIBUTOR} | tr 'A-Z' 'a-z' | cut -d' ' -f1|LC_ALL=C sed 's,[^[:alnum:]_],_,g') ${CLASS}"
 fi
 
diff -dur -x '*.orig' grub-2.00.git20131218.orig/util/grub.d/20_linux_xen.in grub-2.00.git20131218/util/grub.d/20_linux_xen.in
--- grub-2.00.git20131218.orig/util/grub.d/20_linux_xen.in	2013-12-18 14:41:17.000000000 +0100
+++ grub-2.00.git20131218/util/grub.d/20_linux_xen.in	2013-12-18 14:43:11.000000000 +0100
@@ -33,9 +33,9 @@
 CLASS="--class gnu-linux --class gnu --class os --class xen"
 
 if [ "x${GRUB_DISTRIBUTOR}" = "x" ] ; then
-  OS=GNU/Linux
+  OS="$(. /etc/os-release ; echo "$NAME $VERSION")"
 else
-  OS="${GRUB_DISTRIBUTOR} GNU/Linux"
+  OS="${GRUB_DISTRIBUTOR}"
   CLASS="--class $(echo ${GRUB_DISTRIBUTOR} | tr 'A-Z' 'a-z' | cut -d' ' -f1|LC_ALL=C sed 's,[^[:alnum:]_],_,g') ${CLASS}"
 fi
 
