diff -dur grub-2.00.orig/util/grub.d/10_linux.in grub-2.00/util/grub.d/10_linux.in
--- grub-2.00.orig/util/grub.d/10_linux.in	2012-04-18 23:24:38.000000000 +0200
+++ grub-2.00/util/grub.d/10_linux.in	2012-10-27 14:36:58.131180820 +0200
@@ -197,6 +197,7 @@
 	   "initrd-${alt_version}" "initramfs-${alt_version}.img" \
 	   "initramfs-genkernel-${version}" \
 	   "initramfs-genkernel-${alt_version}" \
+	   "initrd-${version}.gz" \
 	   "initramfs-genkernel-${GENKERNEL_ARCH}-${version}" \
 	   "initramfs-genkernel-${GENKERNEL_ARCH}-${alt_version}"; do
     if test -e "${dirname}/${i}" ; then
