--- grub-1.97.1/util/grub-mkconfig.in~	2010-01-12 12:14:40.000000000 +0200
+++ grub-1.97.1/util/grub-mkconfig.in	2010-01-12 12:26:24.581982969 +0200
@@ -29,7 +29,7 @@
 pkgdatadir=${datadir}/`echo @PACKAGE_TARNAME@ | sed "${transform}"`
 grub_prefix=`echo /boot/grub | sed ${transform}`
 grub_cfg=""
-grub_mkconfig_dir=${sysconfdir}/grub.d
+grub_mkconfig_dir=/lib/grub.d
 
 grub_mkdevicemap=${sbindir}/`echo grub-mkdevicemap | sed ${transform}`
 grub_probe=${sbindir}/`echo grub-probe | sed ${transform}`
--- grub-1.99~rc2/conf/Makefile.common~	2011-04-08 13:45:13.000000000 +0200
+++ grub-1.99~rc2/conf/Makefile.common	2011-09-06 22:41:28.748549197 +0200
@@ -98,7 +98,7 @@
 
 # Other variables
 
-grubconfdir = $(sysconfdir)/grub.d
+grubconfdir = /lib/grub.d
 grubdatadir = $(datadir)/`echo @PACKAGE_TARNAME@ | sed '$(transform)'`
 platformdir = $(pkglibrootdir)/$(target_cpu)-$(platform)
 
