diff -dur grub-2.00.orig/conf/Makefile.common grub-2.00/conf/Makefile.common
--- grub-2.00.orig/conf/Makefile.common	2012-10-27 15:05:15.000000000 +0200
+++ grub-2.00/conf/Makefile.common	2012-10-27 15:06:09.135719113 +0200
@@ -98,7 +98,7 @@
 
 # Other variables
 
-grubconfdir = $(sysconfdir)/grub.d
+grubconfdir = /lib/grub.d
 platformdir = $(pkglibdir)
 starfielddir = $(pkgdatadir)/themes/starfield
 
diff -dur grub-2.00.orig/util/grub-mkconfig.in grub-2.00/util/grub-mkconfig.in
--- grub-2.00.orig/util/grub-mkconfig.in	2012-10-27 15:05:15.000000000 +0200
+++ grub-2.00/util/grub-mkconfig.in	2012-10-27 15:06:40.696227008 +0200
@@ -35,7 +35,7 @@
     pkgdatadir="${datadir}/@PACKAGE@"
 fi
 grub_cfg=""
-grub_mkconfig_dir="${sysconfdir}"/grub.d
+grub_mkconfig_dir=/lib/grub.d
 
 self=`basename $0`
 
